source-code/plugins/i18next/src/ideExtension/messageReferenceMatchers.ts

```diff
@@ -8,78 +8,132 @@
-							character: end.column,
+		// Whitespace parser
+		whitespace: () => {
+			return Parsimmon.optWhitespace
+		},
+
+		// Colon parser
+		colon: (r) => {
+			return Parsimmon.string(":").trim(r.whitespace!)
+		},
+
+		// Comma parser
+		comma: (r) => {
+			return Parsimmon.string(",").trim(r.whitespace!)
+		},
+
+		// Parser for namespace value
+		nsValue: (r) => {
+			return Parsimmon.seq(
+				r.whitespace!,
+				Parsimmon.string("ns").trim(r.whitespace!).skip(r.colon!), // namespace key parser
+				r.stringLiteral!.trim(r.whitespace!),
+			).map(([, , val]) => `${val}`)
+		},
+
+		// Parser for t function calls
+		FunctionCall: function (r) {
+			return Parsimmon.seqMap(
+				Parsimmon.regex(/[^a-zA-Z0-9]/), // no preceding letters or numbers
+				Parsimmon.string("t"), // starts with t
+				Parsimmon.string("("), // then an opening parenthesis
+				Parsimmon.index, // start position of the message id
+				r.stringLiteral!, // message id
+				Parsimmon.index, // end position of the message id
+				Parsimmon.regex(/[^)]*/), // ignore the rest of the function call
+				Parsimmon.string(")"), // end with a closing parenthesis
+				(_, __, ___, start, messageId, end, rest) => {
+					const namespaceParser = r
+						.comma! // comma
+						.then(Parsimmon.string("{")) // opening bracket
+						.trim(r.whitespace!) // whitespace
+						.then(r.nsValue!) // namespace value parser
+						.skip(Parsimmon.string("}")) // closing bracket
+						.skip(Parsimmon.regex(/[^)]*/)) // ignore the rest of the function call
+						.trim(r.whitespace!) // whitespace
+
+					const namespace = (namespaceParser.parse(rest) as { status: boolean; value: string })
+						.value
+
+					// -- handle namespaces --
+					// only handle namespaces when the pathPattern require namespaces
+					if (typeof settings.pathPattern === "object") {
+						if (namespace) {
+							// if namespace gets parsed, prepend it to the messageId
+							messageId = namespace + ":" + messageId
+						} else if (!messageId.includes(":")) {
+							// if no namespace gets parsed and the namespace is not already included in the messageId, prepend the default namespace
+							const defaultNamespace = Object.keys(settings.pathPattern)[0]
+							messageId = defaultNamespace + ":" + messageId
+						}
+					}
+
+					return {
+						messageId,
+						position: {
+							start: {
+								line: start.line,
+								character: start.column,
+							},
+							end: {
+								line: end.line,
+								character: end.column,
+							},
 						},
-					},
-				} satisfies MessageReferenceMatch
-			},
-		)
-	},
-})
+					} satisfies MessageReferenceMatch
+				},
+			)
+		},
+	})
+}
 
 // Parse the expression
-export function parse(sourceCode: string): MessageReferenceMatch[] {
+export function parse(sourceCode: string, settings: PluginSettings): MessageReferenceMatch[] {
 	try {
+		const parser = createParser(settings)
 		return parser.entry!.tryParse(sourceCode)
 	} catch {
 		return []
```

Comment:
I didn't add this line. But I am pretty sure that this was on purpose. `tryParse` throws an error when something in the parser goes wrong. Since I can't change the parser, only the `sourceCode` it is hard to produce this on purpose. I would also leave it there just in case something inside throws something.